ArcGISRuntimeSDKQt_CppSamples/Scenes/Add3DTilesLayer/Add3DTilesLayer.cpp

```diff
@@ -0,0 +1,129 @@
+// [WriteFile Name=Add3DTilesLayer, Category=Scenes]
+// [Legal]
+// Copyright 2023 Esri.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// [Legal]
+
+#ifdef PCH_BUILD
+#include "pch.hpp"
+#endif // PCH_BUILD
+
+#include "Add3DTilesLayer.h"
+
+#include "AnalysisListModel.h"
+#include "AnalysisOverlay.h"
+#include "AnalysisOverlayListModel.h"
+#include "ArcGISTiledElevationSource.h"
+#include "Basemap.h"
+#include "Camera.h"
+#include "ElevationSourceListModel.h"
+#include "Layer.h"
+#include "LayerListModel.h"
+#include "LocationLineOfSight.h"
+#include "MapTypes.h"
+#include "Ogc3dTilesLayer.h"
+#include "Point.h"
+#include <PortalItem.h>
+#include "Scene.h"
+#include "SceneQuickView.h"
+#include "SpatialReference.h"
+#include "Surface.h"
+#include "TransformationCatalog.h"
+
+#include <MapView.h>
+#include <QDebug>
+#include <QFuture>
+#include <QStandardPaths>
+#include <QString>
+#include <QUrl>
+#include <QVariantMap>
+#include <QMap>
+
+using namespace Esri::ArcGISRuntime;
+
+const QString DATAPATH = QStandardPaths::writableLocation(QStandardPaths::HomeLocation)+ "/ArcGIS/Runtime/Data/";
+
+Add3DTilesLayer::Add3DTilesLayer(QObject* parent /* = nullptr */):
+  QObject(parent),
+  m_scene(new Scene(BasemapStyle::ArcGISDarkGray, this))
+{
+  // create a new elevation source from Terrain3D REST service
+  ArcGISTiledElevationSource* elevationSource = new ArcGISTiledElevationSource(
+        QUrl("https://elevation3d.arcgis.com/arcgis/rest/services/WorldElevation3D/Terrain3D/ImageServer"), this);
+  // add projection
+  TransformationCatalog::setProjectionEngineDirectory(DATAPATH + "/pedata");
+  if (TransformationCatalog::projectionEngineDirectory().isEmpty())
+    qWarning() << ("Projection Engine Directory not found at: " + DATAPATH + "/pedata");
```

Comment:
Projection engine data isn't mentioned in the design and is no longer required by the API, so I'd say we can remove this.
```suggestion
```